tests/DocumentationExamplesTest.php

```diff
@@ -28,9 +30,14 @@
  * @see https://jira.mongodb.org/browse/DRIVERS-356
  * @see https://jira.mongodb.org/browse/DRIVERS-488
  * @see https://jira.mongodb.org/browse/DRIVERS-547
+ *
+ * Note: Example should use the "inventory" collection in the default database unless they need multiple databases.
```

Comment:
Some thoughts:

The initial `dropCollection()` during `setUp()` is mainly for the benefit of ensuring clean test execution after a previous failure. This correlates with `tearDown()` NOP-ing on a failure, which is more for the benefit of local testing so we inspect a collection after running a failed test in isolation.

Since the test methods in this class are using various databases _and_ collections, I think it makes sense to:

 * Remove the the overridden `getCollectionName()` method in this class
 * Remove the unconditional `dropCollection()` calls in `setUp()` and `tearDown()`.
 * Ensure nothing else in this test is relying on the default collection name

In its place:

 * Allow each test to register any number of databases or individual collections that it will use. For a custom database (as with the transaction examples) you can drop it entirely as you're already doing in this PR. For a collection in the default database, I think it'd make sense to just target the collection directly (since other tests in the suite are sharing the default database).
 * To handle "setUp" behavior, optionally have the registration method drop things or handle that in a second method that you call immediately after. If the dropping is done in a separate method, then you can re-use that in `tearDown()`.
 * The main `setUp()` and `tearDown()` methods should also be responsible for clearing the database/collection registry.

This idea is somewhat related to what I ended up doing with `setUpWithTypeAndRangeOpts()` in [mongodb/mongo-php-library#1064](https://github.com/mongodb/mongo-php-library/pull/1064/files#diff-86fa81787b2fd04d6e360e683890362020e60ca3b3b796570b19aea9ca9ec09bR75), as I needed to customize setup behavior based on the data provider's input.

Reply:
Hum, ok. I suspected those examples to be automatically parsed, but not that much integrated into the doc. I'll implement a more generic mechanism to drop databases or collection invoked by `createCollection` or `dropCollection`.